source/adios2/core/IO.cpp

```diff
@@ -572,6 +572,51 @@ Engine &IO::Open(const std::string &name, const Mode mode, helper::Comm comm)
         }
     }
 
+    // For the inline engine, there must be exactly 1 reader, and exactly 1
+    // writer.
+    if (engineTypeLC == "inline")
+    {
+        if (mode == Mode::Append)
+        {
+            throw std::runtime_error(
+                "Append mode is not supported for the inline engine.");
+        }
+
+        // See inline.rst:44
+        if (mode == Mode::Sync)
```

Comment:
I don't think it does, but to be sure, this doesn't affect single value variables that get read in sync mode by default does it (See `PutSyncCommon` in `InlineWriter.tcc`)? Actually I didn't even realize until now that Sync/Deferred are technically valid options to pass to `IO::Open`. I thought they were just for `Put`/`Get`.

Reply:
Well, they should not be semantically valid options. It looks like Mode is an enum for two sets of options.

Reply:
I was confused about this as well; with Norbert's comments it appears to be a slight inelegance in the code. I'll therefore also prohibit opening an engine in deferred mode in a separate PR since I don't want to get the commit graph all messed up for #2392.

Reply:
So should the check for mode being Sync or Deferred be done for all engines instead of only inline?

Reply:
@caitlinross: Good question; it seems to me that it should be for all engines, but I'll leave @pnorbert to make that decision.